wikipedia_cleanup/schema.py

```diff
@@ -40,12 +40,28 @@ class InfoboxChange(BaseModel):
     revision_valid_to: Optional[
         datetime
     ] = None  # date of the next revision for that infobox
+
     # => revision_valid_to <= value_valid_to
 
     def __str__(self) -> str:
         return str(self.__dict__)
 
 
+"""
+Empty class that can be used to use less RAM. All attributes are basically optional.
+But optional would still cost us a nullptr.
+Therefore all attributes need to be added on manually.
+E.g.
+a = SparseInfoboxChange()
+a.page_id = change.page_id
+...
+"""
+
+
+class SparseInfoboxChange:
+    pass
```

Comment:
we're running into a nasty tradeoff here. We want to be able to access the changes in the same way and thus a class makes sense, but having both these related change classes without inheritance will make changing one or the other a lot more difficult. I guess for now its ok, but this might prove to be a problem later

Reply:
Yes indeed this might be the case